src/sentry/api/endpoints/organization_repository_details.py

```diff
@@ -65,6 +65,19 @@ def delete(self, request, organization, repo_id):
         except Repository.DoesNotExist:
             raise ResourceDoesNotExist
 
+        # if repo doesn't have commits, delete immediately
+        has_commits = Commit.objects.filter(
+            repository_id=repo.id,
+            organization_id=organization.id,
+        ).exists()
+        if not has_commits:
+            repo.get_provider().delete_repository(
```

Comment:
k .. yeah kind of though about that as i was putting pr up lol :) easy to fix though